contracts/yield-source/ATokenYieldSource.sol

```diff
@@ -158,17 +165,18 @@ contract ATokenYieldSource is ERC20Upgradeable, IProtocolYieldSource, AssetManag
   function supplyTokenTo(uint256 mintAmount, address to) external override {
```

Comment:
I believe this needs a reentrancy guard as well:

Re-entrancy attack:

User has existing shares.

1. User calls `supplyTokenTo`
2. In the call to `_mint`: after tokens have been transferred to lending pool, re-enter into `redeemToken`
3. Redeem will pull out old shares, which have a higher value because of the new assets.
4. `supplyTokenTo` resumes, and mints shares to the user.
5. User redeems the new shares in a separate tx.

Numbers:

User has 10 shares out of a total of 100.  Total assets held by the contract is 1000 Dai.

1. User calls `supplyTokenTo` with 500 Dai.  This will mint them 50 shares.
2. In the call to `_mint` the user re-enters `redeemToken` after the Dai is transferred.
3. `redeemToken` redeems their olds shares: 10/100 of the 1500 Dai = 150 Dai.
4. `supplyTokenTo` resumes, and mints 50 shares to user.
5. User redeems 50 shares from 1350 Dai, or 675 Dai.

Total deposited: 600 Dai.  Total withdrawn: 825 Dai

Reply:
Done! avg gas goes from ~137k to~159k (as expected)

Reply:
Ouch